vernac/vernacextend.ml

```diff
@@ -36,7 +36,7 @@ type vernac_classification =
      don't modify the interpretation state. *)
   | VtQuery
   (* Commands that change the current proof mode *)
-  | VtProofMode of string
+  | VtProofMode of Pvernac.proof_mode option
```

Comment:
```suggestion
  | VtProofMode of Pvernac.proof_mode
```
you error when the option is empty, so get the value out

Reply:
I thought about that, but in stm, VCS.new_node expects an option and that routine is called a number of places.  So it doesn't simplify anything.

```
      | VtProofMode proof_mode ->
        let id = VCS.new_node ~id:newtip proof_mode () in
```

Reply:
It is not about simplification but semantics
None as returned by lookup_proof_mode means it was incorrect
None as taken by new_node means the command doesn't set the proof mode

Reply:
This. Enforce impossible cases by typing, not runtime checks.

Reply:
OK, fixed.